### ISSUE TITLE
Readthedocs build failure

### ISSUE THREAD

**Issue by adswa** [datalad/datalad#6122]
The [readthedocs builds](https://readthedocs.org/projects/datalad/builds/15127344/) currently fail:

```
Running Sphinx v1.8.5
running build_manpage
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
running build_cfginfo
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
running build_schema
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
schema written to /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/_extras/schema_v2.0.json
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
loading translations [en]... done
making output directory...
[autosummary] generating autosummary for: acknowledgements.rst, background.rst, basics.rst, changelog.rst, cmdline.rst, config.rst, customization.rst, design/credentials.rst, design/dataset_argument.rst, design/drop.rst, ..., generated/man/datalad-unlock.rst, generated/man/datalad-update.rst, generated/man/datalad-wtf.rst, generated/man/datalad.rst, glossary.rst, index.rst, metadata.rst, modref.rst, publications.rst, related.rst
[autosummary] generating autosummary for: /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.Dataset.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.add_archive_content.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.add_readme.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.addurls.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.aggregate_metadata.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.annotate_paths.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.check_dates.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.clean.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.clone.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.api.copy_file.rst, ..., /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.customremotes.main.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.log.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.support.annexrepo.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.support.archives.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.support.gitrepo.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.tests.heavyoutput.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.tests.utils.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.tests.utils_testrepos.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.utils.rst, /home/docs/checkouts/readthedocs.org/user_builds/datalad/checkouts/latest/docs/source/generated/datalad.version.rst
loading intersphinx inventory from http://handbook.datalad.org/en/latest/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 69 source files that are out of date
updating environment: 130 added, 0 changed, 0 removed
reading sources... [  0%] acknowledgements
reading sources... [  1%] background
reading sources... [  2%] basics

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/cmd/build.py", line 304, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 360, in build
    updated_docnames = set(self.read())
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 468, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 490, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/builders/__init__.py", line 534, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/io.py", line 318, in read_doc
    pub.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/docutils/core.py", line 219, in publish
    self.apply_transforms()
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/docutils/core.py", line 200, in apply_transforms
    self.document.transformer.apply_transforms()
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/transforms/__init__.py", line 90, in apply_transforms
    Transformer.apply_transforms(self)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/transforms/__init__.py", line 245, in apply
    apply_source_workaround(n)
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/util/nodes.py", line 94, in apply_source_workaround
    for classifier in reversed(node.parent.traverse(nodes.classifier)):
TypeError: 'generator' object is not reversible

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/datalad/envs/latest/lib/python3.7/site-packages/sphinx/util/nodes.py", line 94, in apply_source_workaround
    for classifier in reversed(node.parent.traverse(nodes.classifier)):
TypeError: 'generator' object is not reversible
The full traceback has been saved in /tmp/sphinx-err-074_uni5.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

This is a [known problem](https://github.com/readthedocs/readthedocs.org/issues/8616), and appearently due to a new release of docutils. The suggested fix is to pin ``docutils<0.18``.

**Comment by mih** [comment]
So indeed we get

```
Collecting sphinx<2
  Downloading Sphinx-1.8.5-py2.py3-none-any.whl (3.1 MB)
```

Likely because of https://github.com/readthedocs/readthedocs.org/issues/8616#issuecomment-952982815, leading to https://docs.readthedocs.io/en/stable/builds.html?highlight=october%202020#python

> Projects created before Oct 20, 2020 use 1.8.x. New projects use the latest version.

So I think we should be forcing a later version. Not yet sure how.

**Comment by mih** [comment]
Did not work. For some reason RTD does not use `requirements-devel.txt`, although I have enabled that it should inspect it.

**Comment by adswa** [comment]
could you add me (I should be "AdinaWagner") as a maintainer on readthedocs? I could try, too, then

**Comment by yarikoptic** [comment]
done

**Comment by adswa** [comment]
Fixed with 7631474. I believe that the failure to see PR previews stems from the fact that we PR from forks, and we wouldn't be able to get them unless we push to upstream.
